docs/src/2.3.RenderingObjects.mdx

```diff
@@ -0,0 +1,196 @@
+
+```js
+// src/Example.js
+function Example() {
+  const steps = 500; // total amount of objects
+
+  // map a number/index to a specific color
+  function numberToColor(number, max, a = 1) {
+    const i = (number * 255) / max;
+    const r = Math.round(Math.sin(0.024 * i + 0) * 127 + 128) / 255;
+    const g = Math.round(Math.sin(0.024 * i + 2) * 127 + 128) / 255;
+    const b = Math.round(Math.sin(0.024 * i + 4) * 127 + 128) / 255;
+    return { r, g, b, a };
+  }
+
+  // the object index needs to be multipled by this scale so it's evenly distributed in the space
+  const scale = (Math.PI * 2) / steps;
+  const markers = new Array(steps)
+    .fill()
+    .map((_, idx) => [
+      // generate x, y, z coordinates based on trefoil equation
+      Math.sin(idx * scale) + 2 * Math.sin(2 * idx * scale),
+      Math.cos(idx * scale) - 2 * Math.cos(2 * idx * scale),
+      -Math.sin(3 * idx * scale),
+    ])
+    .map(([x, y, z], idx) => ({
+      pose: {
+        orientation: { x: 0, y: 0, z: 0, w: 1 },
+        position: { x: 20 * x, y: 20 * y, z: 20 * z },
+      },
+      scale: { x: 1, y: 1, z: 1 },
+      color: numberToColor(idx, steps),
+    }));
+
+  return (
+    <Worldview>
+      <Spheres>{markers}</Spheres>
+      <Axes />
+    </Worldview>
+  );
+}
+```
+
+<CodeSandboxEmbed src="https://codesandbox.io/s/2x26ov87j0" />
+
+## Improving Performance with Instanced Rendering
+
+When drawing many instances of the same object, you'll quickly reach a performance bottleneck because of amount of drawing calls on the GPU. Instancing is a technique where we draw many objects at once with a single render call, save us all the CPU -> GPU communication time we need to render each object. Learn more about [instancing](https://learnopengl.com/Advanced-OpenGL/Instancing).
+
+Most of the existing [Command](#/docs/api/command) components support instancing. In this step, we are going to convert previous example to use instanced rendering.
+
+```js
+// src/Example.js
+function Example() {
+  // ... other old code
+  new Array(steps)
+    .fill()
+    .map((_, idx) => [
+      // generate x, y, z coordinates based on trefoil equation
+      Math.sin(idx * scale) + 2 * Math.sin(2 * idx * scale),
+      Math.cos(idx * scale) - 2 * Math.cos(2 * idx * scale),
+      -Math.sin(3 * idx * scale),
+    ])
+    .forEach(([x, y, z], idx) => {
+      // add individual point and color to the single sphere object
+      marker.colors.push(numberToColor(idx, steps));
+      marker.points.push({ x: x * 20, y: y * 20, z: z * 20 });
+    });
+
+  return (
+    <Worldview>
+      <Spheres>{[marker]}</Spheres>
+      <Axes />
+    </Worldview>
+  );
+}
+```
+
+Notice in the updated example, instead of rendering 500 markers, we rendered a single marker with 500 colors and points? It's a very simple steps, but it saved lots of CPU -> GPU time!
```

Comment:
'It's a very simple step'

Reply:
we may also want to emphasize that in this example we used the `colors` and `points` properties.